Google.PowerShell/Common/ActiveUserCredential.cs

```diff
@@ -0,0 +1,172 @@
+*/
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Flows;
+using Google.Apis.Http;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.PowerShell.Common
+{
+    /// <summary>
+    /// OAuth 2.0 credential for accessing protected resources using an access token.
+    /// This class will get the access token from "gcloud auth print-access-token".
+    /// </summary>
+    public class ActiveUserCredential : ICredential, IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler
+    {
+        private static ActiveUserToken token;
+        private static object lockObject = new object();
+
+        /// <summary>
+        /// We just need a default flow for things like flow.AccessMethod.Intercept
+        /// or flow.RevokeTokenAsync.
+        /// </summary>
+        private readonly IAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow(
+                new GoogleAuthorizationCodeFlow.Initializer()
+                {
+                    ClientSecrets = new ClientSecrets()
+                    {
+                        ClientId = "clientId",
+                        ClientSecret = "clientSecrets"
+                    }
+                });
+
+        /// <summary>UserId is not important, it can be anything.</summary>
+        private readonly string userId = "userId";
+
+        /// <summary>Gets or sets the token response which contains the access token.</summary>
+        private static ActiveUserToken Token
+        {
+            get
+            {
+                lock (lockObject)
+                {
+                    return token;
+                }
+            }
+            set
+            {
+                lock (lockObject)
+                {
+                    token = value;
+                }
+            }
+        }
+
+        /// <summary>Gets the authorization code flow.</summary>
+        public IAuthorizationCodeFlow Flow
+        {
+            get { return flow; }
+        }
+
+        /// <summary>Constructs a new credential instance.</summary>
+        public ActiveUserCredential()
+        {
+            if (Token == null)
+            {
+                Token = GCloudWrapper.GetAccessToken().Result;
+            }
+        }
+
+        #region IHttpExecuteInterceptor
+
+        /// <summary>
+        /// Default implementation is to try to refresh the access token if there is no access token or if we are 1 
+        /// minute away from expiration. If token server is unavailable, it will try to use the access token even if 
+        /// has expired. If successful, it will call <see cref="IAccessMethod.Intercept"/>.
+        /// </summary>
+        public async Task InterceptAsync(HttpRequestMessage request, CancellationToken taskCancellationToken)
+        {
+            var accessToken = await GetAccessTokenForRequestAsync(request.RequestUri.ToString(), taskCancellationToken).ConfigureAwait(false);
```

Comment:
Please introduce some local variables. It is unclear what type `accessToken` has. Perhaps:

```
Foo x = GetAccessTokenForRequestAsync(request.RequestUri.ToString(), taskCancellationToken);
Bar accessToken = x.ConfigureAwait(false);
````

What do you think?